sklearn/tests/test_impute.py

```diff
@@ -309,3 +313,215 @@ def test_imputation_copy():
+        mice = MICEImputer(missing_values=0,
+                           n_imputations=1,
+                           n_burn_in=1,
+                           initial_strategy=strategy).fit(X_train)
+        initial_imputer = SimpleImputer(missing_values=0,
+                                        strategy=strategy).fit(X_train)
+
+        # if there were no missing values at time of fit, then mice will
+        # only use the initial imputer for that feature at transform
+        assert np.all(mice.transform(X_test)[:, 0] ==
+                      initial_imputer.transform(X_test)[:, 0])
+
+
+def test_mice_transform_stochasticity():
+    rng = np.random.RandomState(0)
+    n = 100
+    d = 10
+    X = sparse_random_matrix(n, d, density=0.10,
+                             random_state=rng).toarray()
+
+    imputer = MICEImputer(missing_values=0,
+                          n_imputations=1,
+                          n_burn_in=1)
+    imputer.fit(X)
+
+    X_fitted_1 = imputer.transform(X)
+    X_fitted_2 = imputer.transform(X)
+
+    # sufficient to assert that the means are not the same
+    assert_not_equal(np.mean(X_fitted_1), np.mean(X_fitted_2))
+
+
+def test_mice_no_missing():
+    rng = np.random.RandomState(0)
+    X = rng.rand(100, 100)
+    X[:, 0] = np.nan
+    m1 = MICEImputer(n_imputations=10)
+    m2 = MICEImputer(n_imputations=10)
+    pred1 = m1.fit(X).transform(X)
+    pred2 = m2.fit_transform(X)
+    # should exclude the first column entirely
+    assert_array_almost_equal(X[:, 1:], pred1)
+    # fit and fit_transform should both be identical
+    assert_array_almost_equal(pred1, pred2)
+
+
+def test_mice_transform_recovery():
+    rng = np.random.RandomState(0)
+
+    def make_data(rank):
+        n = 100
+        d = 100
+        A = rng.rand(n, rank)
+        B = rng.rand(rank, d)
+        Xfilled = np.dot(A, B)
+        # half is randomly missing
+        nan_mask = rng.rand(n, d) < 0.5
+        X_missing = Xfilled.copy()
+        X_missing[nan_mask] = np.nan
+
+        # split up data in half
+        n = n // 2
+        X_train_filled = Xfilled[:n]
+        X_train = X_missing[:n]
+        X_test_filled = Xfilled[n:]
+        X_test = X_missing[n:]
+        return X_train_filled, X_train, X_test_filled, X_test
+
+    for rank in [5, 10]:
+        X_train_filled, X_train, X_test_filled, X_test = make_data(rank)
+        imputer = MICEImputer(n_imputations=10,
+                              n_burn_in=10,
+                              verbose=True).fit(X_train)
+        X_test_est = imputer.transform(X_test)
+        assert_array_almost_equal(X_test_filled, X_test_est, decimal=1)
+
+
+def test_mice_additive_matrix():
+    rng = np.random.RandomState(0)
+    n = 100
+    d = 10
+    A = rng.randn(n, d)
+    B = rng.randn(n, d)
+    Xfilled = np.zeros(A.shape)
+    for i in range(d):
+        for j in range(d):
+            Xfilled[:, (i+j) % d] += (A[:, i] + B[:, j]) / 2
```

Comment:
use `from __future__ import division` on the top

Reply:
Will do.